apps/src/blockly/addons/functionEditor.js

```diff
@@ -142,10 +142,14 @@ export default class FunctionEditor {
       const existingData = Blockly.serialization.blocks.save(
         existingProcedureBlock
       );
+      // Disable events here so we don't copy an existing block into the hidden definition
+      // workspace.
+      Blockly.Events.disable();
```

Comment:
@ebeastlake as I was doing clean up here I realized this is the culprit for the extra procedures in the hidden workspace, so disabling events here should cover that issue.

Reply:
Nice find!